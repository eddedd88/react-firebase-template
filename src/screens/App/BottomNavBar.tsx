import React from 'react'
import BottomNavigation from '@material-ui/core/BottomNavigation'
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction'
import { Link, useLocation } from 'react-router-dom'
import PersonIcon from '@material-ui/icons/Person'
import LocationCityIcon from '@material-ui/icons/LocationCity'
import routes from '../routes'
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles'

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      position: 'fixed',
      bottom: 0,
      width: '100%',
      boxShadow: theme.shadows[8],
      backfaceVisibility: 'hidden'
    }
  })
)

export const BottomNavBar = () => {
  const classes = useStyles()
  const location = useLocation()

  const secondPathIndex = location.pathname.indexOf('/', 1)
  const tabValue =
    secondPathIndex < 0
      ? location.pathname
      : location.pathname.substring(0, secondPathIndex)

  return (
    <BottomNavigation value={tabValue} classes={classes} showLabels>
      <BottomNavigationAction
        icon={<LocationCityIcon />}
        label='Home'
        value={routes.home}
        component={Link}
        to={routes.home}
      />
      <BottomNavigationAction
        icon={<PersonIcon />}
        label='Profile'
        value={routes.signin}
        component={Link}
        to={routes.signin}
      />
    </BottomNavigation>
  )
}

export default BottomNavBar
