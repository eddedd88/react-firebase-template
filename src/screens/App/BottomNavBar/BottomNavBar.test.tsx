import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import BottomNavBar from './BottomNavBar'

jest.mock('react-router-dom', () => ({
  useLocation: () => ({
    pathname: '/test'
  })
}))

describe('BottomNavBar', () => {
  it('renders correctly', () => {
    const renderer = ShallowRenderer.createRenderer()
    const tree = renderer.render(<BottomNavBar />)
    expect(tree).toMatchSnapshot()
  })
})
