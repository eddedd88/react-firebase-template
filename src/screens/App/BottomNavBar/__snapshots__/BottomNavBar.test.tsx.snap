// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavBar renders correctly 1`] = `
<ForwardRef(WithStyles)
  classes={
    Object {
      "root": "makeStyles-root-1",
    }
  }
  showLabels={true}
  value="/test"
>
  <ForwardRef(WithStyles)
    icon={<Memo(LocationCityIcon) />}
    label="Feed"
    to="/feed"
    value="/feed"
  />
  <ForwardRef(WithStyles)
    icon={<Memo(GridOnIcon) />}
    label="Grid"
    to="/grid"
    value="/grid"
  />
  <ForwardRef(WithStyles)
    icon={<Memo(PersonIcon) />}
    label="Profile"
    to="/profile"
    value="/profile"
  />
</ForwardRef(WithStyles)>
`;
